scripts/encodings_retrival/extract_geneformer_encodings.py

```diff
@@ -0,0 +1,122 @@
+import pickle
+import tempfile
+from pathlib import Path
+
+import click
+import pandas as pd
+import requests
+import torch
+
+GENEFORMER_URL = "https://huggingface.co/ctheodoris/Geneformer/resolve/main/pytorch_model.bin?download=true"
+TOKEN_DICT_URL = "https://huggingface.co/ctheodoris/Geneformer/resolve/main/geneformer/token_dictionary.pkl?download=true"
+
+
+def download_file_from_url(url, file_name, dir_path, chunk_size=8192):
+    """
+    Download a file fom a url path.
+
+    Args:
+    ----
+        url (str): url of target file
+        file_name (str): name of output file
+        dir_path (pathlib.Path): path to dir for saving the file
+        chunk_size (int): size of data to be streamed
+
+    Returns:
+    -------
+        The path of the saved file
+
+    """
+    local_filename = dir_path / file_name
+    with requests.get(url, stream=True) as response:
+        response.raise_for_status()
+        with open(local_filename, "wb") as file:
+            for chunk in response.iter_content(chunk_size=chunk_size):
+                file.write(chunk)
+    return local_filename
+
+
+def add_tokens_as_index(encodings, token_dict):
+    """
+    Add tokens names to the encodings file.
+
+    Args:
+    ----
+        encodings (torch.tensor): the encodings
+        token_dict (dict): mapping of symbol name to index
+
+    Returns:
+    -------
+        encodings with the symbols
+
+    """
+    encodings_with_symbols = pd.DataFrame(
+        encodings["bert.embeddings.word_embeddings.weight"].detach(),
+        index=list(token_dict.keys()),
+    )
+    return encodings_with_symbols
+
+
+def save_encodings(encodings, output_file_dir):
+    """
+    Save the gene encodings to the output dir.
+
+    Args:
+    ----
+        encodings (pd.DataFrame): the encodings
+        output_file_dir (str): path to dir for saving the file
+
+    """
+    model_encodings_dir = Path(output_file_dir) / "Geneformer"
+    model_encodings_dir.mkdir(parents=True, exist_ok=True)
+    output_file_path = model_encodings_dir / "encodings.csv"
```

Comment:
Same here; "encodings.csv" and "symbol" should be variabels in the method header with this values as default.